code/_onclick/hud/parallax.dm

```diff
@@ -0,0 +1,254 @@
+/client
+	var/list/parallax_layers = list()
+	var/obj/screen/parallax_pmaster/parallax_pmaster
+	var/obj/screen/parallax_space_whitifier/parallax_space_whitifier
+	var/obj/screen/parallax_fixer/parallax_fixer
+	var/atom/movable/movingmob
+	var/turf/previous_turf
+	var/do_smoothing = 1
+	var/looping_mode = 0
+	var/last_parallax_shift
+	var/parallax_movedir = 0
+
+/datum/hud/proc/create_parallax()
+	var/client/C = mymob.client
+	if(!C.parallax_layers.len)
+		C.parallax_layers += new /obj/screen/parallax_layer/layer_1
+		C.parallax_layers += new /obj/screen/parallax_layer/layer_2
+		C.parallax_pmaster = new
+		C.parallax_space_whitifier = new
+		C.parallax_fixer = new
+	for(var/obj/O in C.parallax_layers)
+		C.screen |= O
+		C.screen |= C.parallax_pmaster
+		C.screen |= C.parallax_space_whitifier
+		C.screen |= C.parallax_fixer
```

Comment:
when you modify special appearance related lists like client images, overlays, or screen, you should try to keep the number of actual modifications low. replace this with C.screen |= (O | C.parallax_pmaster | C.parallax_space_whitier | C.parallax_fixer)

Byond does stupid calculate on write shit when you write to special lists.